### PR TITLE
drivers/tmp00x : Expose Configurations to Kconfig 

### PR DESCRIPTION
**Pull request by akshaim**
### Contribution description

This PR exposes compile configurations in TMP006/TMP007 Sensor Device driver to Kconfig.

### Testing procedure
1. New documentation was built using doxygen 

The build worked fine.

2. New macro was introduced in tests/driver_tmp00x/main.c for testing.

```
#define STR(x)   #x
#define SHOW_DEFINE(x) printf("%s=%s\n", #x, STR(x))
```
Firmware was uploaded to FIT IoT Lab test bed.

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-309-g0af78-Kconfig_tmp00x_1_tests)
CONFIG_TMP00X_I2C_ADDRESS=(0x40)
CONFIG_TMP00X_CONVERSION_TIME=(1E6)
CONFIG_TMP00X_USE_LOW_POWER=CONFIG_TMP00X_USE_LOW_POWER
CONFIG_TMP00X_USE_RAW_VALUES=CONFIG_TMP00X_USE_RAW_VALUES
TMP00X infrared thermopile sensor driver test application

Initializing TMP00X sensor at I2C_0 ... tmp00x_init: error reading device ID!
init device [ERROR]

#### Usage with CFLAGS 

/tests/driver_tmp00x/Makefile

> CFLAGS += -DCONFIG_TMP00X_I2C_ADDRESS=0x41
> CFLAGS += -DCONFIG_TMP00X_CONVERSION_TIME=2E6
> CFLAGS += -DCONFIG_TMP00X_USE_LOW_POWER=1
> CFLAGS += -DCONFIG_TMP00X_USE_RAW_VALUES=1

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-310-gba881-Kconfig_tmp00x_1_tests)
CONFIG_TMP00X_I2C_ADDRESS=0x41
CONFIG_TMP00X_CONVERSION_TIME=2E6
CONFIG_TMP00X_USE_LOW_POWER=1
CONFIG_TMP00X_USE_RAW_VALUES=1
TMP00X infrared thermopile sensor driver test application

Initializing TMP00X sensor at I2C_0 ... tmp00x_init: error reading device ID!
init device [ERROR]

#### Usage with Kconfig

/tests/driver_tmp00x/

> make menuconfig

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-310-ga6e69-Kconfig_tmp00x_1_tests)
CONFIG_TMP00X_I2C_ADDRESS=0x45
CONFIG_TMP00X_CONVERSION_TIME=(25E4)
CONFIG_TMP00X_USE_LOW_POWER=1
CONFIG_TMP00X_USE_RAW_VALUES=1
TMP00X infrared thermopile sensor driver test application

Initializing TMP00X sensor at I2C_0 ... tmp00x_init: error reading device ID!
init device [ERROR]

Note : The sensor is not available for interfacing hence configurability of macros were only tested.

### Issues/PRs references

#12888
@leandrolanzieri  Please merge this only after  #13981

### CODE REVIEW

File: drivers/include/tmp00x.h [hunk 1]
```diff
@@ -100,40 +101,59 @@ extern "C"
 /**
  * @brief   Default Address
  *
- * I2C Address depends on the state of ADR0 and ADR1 Pins
- * For more information, please refer to section 7.3.6.2 of TMP007 datasheet (SBOS685B).
+ * TMP006/TMP007 allows for upto 8 devices on a single bus.
```
Comment (leandrolanzieri) [inline drivers/include/tmp00x.h:104]:
```suggestion
 * TMP006/TMP007 allows for up to 8 devices on a single bus.
```

File: drivers/include/tmp00x.h [hunk 2]
```diff
@@ -100,40 +101,59 @@ extern "C"
...
+#ifndef CONFIG_TMP00X_I2C_ADDRESS
+#define CONFIG_TMP00X_I2C_ADDRESS         (0x40)
 #endif
 
 /**
  * @brief   Default Conversion Time in us
+ * 
+ * The duration of the analog-to-digital(A/D) conversion is
+ * determined by the conversion rate bits CR0, CR1 and CR2.
+ * For more information refer datasheet.
```
Comment (leandrolanzieri) [inline drivers/include/tmp00x.h:119]:
```suggestion
 * For more information refer to the datasheet.
```

File: drivers/tmp00x/Kconfig [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        Configure the TMP00X driver using Kconfig.
+
+if KCONFIG_MODULE_TMP00X
+
+config TMP00X_I2C_ADDRESS
+    hex "Default I2C Address"
+    range 0x40 0x47
+    default 0x40
+    help
+        TMP006/TMP007 allows for upto 8 devices on a single bus.
```
Comment (leandrolanzieri) [inline drivers/tmp00x/Kconfig:20]:
```suggestion
        TMP006/TMP007 allows for up to 8 devices on a single bus.
```

File: drivers/tmp00x/Kconfig [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        The address value depends on the state of ADR0 and ADR1
+        pins. Default value (0x40) corresponds to ADR0 and ADR1
+        pins tied to GND. For more information refer Serial Bus
+        Address section in Datasheet.
+
```
Comment (leandrolanzieri) [inline drivers/tmp00x/Kconfig:25]:
```suggestion
        pins tied to GND. For more information refer to the 'Serial Bus
        Address' section in the datasheet.

```

File: drivers/include/tmp00x.h [hunk 5]
```diff
@@ -100,40 +101,59 @@ extern "C"
 /**
  * @brief   Default Address
  *
- * I2C Address depends on the state of ADR0 and ADR1 Pins
- * For more information, please refer to section 7.3.6.2 of TMP007 datasheet (SBOS685B).
+ * TMP006/TMP007 allows for upto 8 devices on a single bus.
+ * The address value depends on the state of ADR0 and ADR1
+ * pins. Default value (0x40) corresponds to ADR0 and ADR1
+ * pins tied to GND. For more information refer Serial Bus
+ * Address section in Datasheet.
```
Comment (leandrolanzieri) [inline drivers/include/tmp00x.h:108]: Same as in Kconfig file

File: drivers/tmp00x/Kconfig [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+choice
+    bool "Conversion time"
+    default TMP00X_CONVERSION_TIME_1s
+    help
+        The duration of the analog-to-digital(A/D) conversion is
+        determined by the conversion rate bits CR0, CR1 and CR2.
+        For more information refer datasheet.
+
+config TMP00X_CONVERSION_TIME_0_25s
```
Comment (leandrolanzieri) [inline drivers/tmp00x/Kconfig:34]: Let's keep all uppercase (`0_25S`)

Comment (akshaim) [reply drivers/tmp00x/Kconfig]: Done. Updated

**Comment by leandrolanzieri** [comment]
Please rebase

**Comment by akshaim** [comment]
> Please rebase

Done.

File: drivers/tmp00x/tmp00x.c [hunk 7]
```diff
@@ -202,26 +204,26 @@ int tmp00x_read_temperature(const tmp00x_t *dev, int16_t *ta, int16_t *to)
 {
 
     uint16_t drdy;
-#if (!TMP00X_USE_RAW_VALUES)
+#if (!IS_ACTIVE(CONFIG_TMP00X_USE_RAW_VALUES))
```
Comment (leandrolanzieri) [inline drivers/tmp00x/tmp00x.c:207]: I think these conditionals can be C `if`.

Comment (akshaim) [reply drivers/tmp00x/tmp00x.c:207]: I did try it but it returned error.

File: drivers/tmp00x/Kconfig [hunk 8]
```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2020 Freie Universitaet Berlin
```
[the remark this answers is not in the document]
Comment (akshaim) [reply drivers/tmp00x/Kconfig:1]: Strange. Was using an auto wrapping plugin. Corrected now. Thanks for pointing it out.

File: drivers/include/tmp00x.h [hunk 9]
```diff
@@ -100,40 +101,59 @@ extern "C"
...
  */
-#ifndef TMP00X_I2C_ADDRESS
-#define TMP00X_I2C_ADDRESS         (0x40)
+#ifndef CONFIG_TMP00X_I2C_ADDRESS
+#define CONFIG_TMP00X_I2C_ADDRESS         (0x40)
 #endif
 
 /**
  * @brief   Default Conversion Time in us
+ *
```
[the remark this answers is not in the document]
Comment (akshaim) [reply drivers/include/tmp00x.h:116]: Thanks. Updated.

File: drivers/tmp00x/tmp00x.c (hunk 7 above)
Comment (leandrolanzieri) [reply drivers/tmp00x/tmp00x.c:207]: Probably you need to also change the conditional include of `xtimer.h`

Comment (akshaim) [reply drivers/tmp00x/tmp00x.c]: Done. Updated and tested.

File: drivers/tmp00x/tmp00x.c [hunk 10]
```diff
@@ -200,54 +200,50 @@ void tmp00x_convert(int16_t rawv, int16_t rawt,  float *tamb, float *tobj)
...
+    else {
+        if ((ret = tmp00x_read(dev, &rawvolt, &rawtemp, &drdy)) < 0) {
+            return ret;
+        }
+
+        if (!drdy) {
+            if (IS_ACTIVE(CONFIG_TMP00X_USE_LOW_POWER)) {
+                tmp00x_set_standby(dev);
+            }
         return -TMP00X_ERROR;
```
Comment (leandrolanzieri) [inline drivers/tmp00x/tmp00x.c:236]: Same here

File: drivers/tmp00x/tmp00x.c [hunk 11]
```diff
@@ -200,54 +200,50 @@ void tmp00x_convert(int16_t rawv, int16_t rawt,  float *tamb, float *tobj)
...
-#if TMP00X_USE_LOW_POWER
-        tmp00x_set_standby(dev);
-#endif
-        return -TMP00X_ERROR;
-    }
-
+        }
     tmp00x_convert(rawvolt, rawtemp,  &tamb, &tobj);
     *ta = (int16_t)(tamb*100);
     *to = (int16_t)(tobj*100);
```
Comment (leandrolanzieri) [inline drivers/tmp00x/tmp00x.c:240]: Same here

**Comment by leandrolanzieri** [comment]
Result of the driver test application:
```
2020-05-18 14:44:25,454 # a�main(): This is RIOT! (Version: 2020.07-devel-523-g5f191-Kconfig_tmp00x_1)
2020-05-18 14:44:25,459 # TMP00X infrared thermopile sensor driver test application
2020-05-18 14:44:25,463 # 
2020-05-18 14:44:29,461 # Initializing TMP00X sensor at I2C_0 ... [SUCCESS]
2020-05-18 14:44:29,461 # 
2020-05-18 14:44:29,465 # Raw data T:  3056   V:   -43
2020-05-18 14:44:29,469 # Data Tabm: 2387   Tobj: 2711
2020-05-18 14:44:33,469 # Raw data T:  3052   V:   106
2020-05-18 14:44:33,472 # Data Tabm: 2384   Tobj: 3039
2020-05-18 14:44:37,471 # Raw data T:  3048   V:   110
2020-05-18 14:44:37,474 # Data Tabm: 2381   Tobj: 3045
2020-05-18 14:44:41,474 # Raw data T:  3044   V:   103
2020-05-18 14:44:41,477 # Data Tabm: 2378   Tobj: 3026
2020-05-18 14:44:45,476 # Raw data T:  3044   V:   120
2020-05-18 14:44:45,479 # Data Tabm: 2378   Tobj: 3063
2020-05-18 14:44:49,479 # Raw data T:  3036   V:   -70
2020-05-18 14:44:49,483 # Data Tabm: 2371   Tobj: 2633
```
I changed the conversion time to 4 seconds via `menuconfig`

File: drivers/tmp00x/tmp00x.c (hunk 11 above)
Comment (akshaim) [reply drivers/tmp00x/tmp00x.c]: Updated.

File: drivers/tmp00x/tmp00x.c [hunk 12]
```diff
@@ -200,54 +200,50 @@ void tmp00x_convert(int16_t rawv, int16_t rawt,  float *tamb, float *tobj)
...
+    }
+
+    if (IS_ACTIVE(CONFIG_TMP00X_USE_RAW_VALUES)) {
+        if ((ret = tmp00x_read(dev, to, ta, &drdy)) < 0) {
+            return ret;
+        }
+
+        if (!drdy) {
+            if (IS_ACTIVE(CONFIG_TMP00X_USE_LOW_POWER)) {
+            tmp00x_set_standby(dev);
```
[the remark this answers is not in the document]
Comment (akshaim) [reply drivers/tmp00x/tmp00x.c]: Done

File: drivers/saul/init_devs/auto_init_tmp00x.c [hunk 13]
```diff
@@ -69,12 +70,12 @@ void auto_init_tmp00x(void)
             LOG_ERROR("[auto_init_saul] error set active tmp00x #%u\n", i);
             continue;
         }
-#if TMP00X_USE_LOW_POWER
-        if (tmp00x_set_standby(&tmp00x_devs[i]) != TMP00X_OK) {
-            LOG_ERROR("[auto_init_saul] error set standby tmp00x #%u\n", i);
-            continue;
-        }
-#endif
+    if (IS_ACTIVE(CONFIG_TMP00X_USE_LOW_POWER)) {
```
[the remark this answers is not in the document]
Comment (akshaim) [reply drivers/saul/init_devs/auto_init_tmp00x.c]: Done

**Comment by akshaim** [comment]
> Result of the driver test application:
> 
> ```
> 2020-05-18 14:44:25,454 # a�main(): This is RIOT! (Version: 2020.07-devel-523-g5f191-Kconfig_tmp00x_1)
> 2020-05-18 14:44:25,459 # TMP00X infrared thermopile sensor driver test application
> 2020-05-18 14:44:25,463 # 
> 2020-05-18 14:44:29,461 # Initializing TMP00X sensor at I2C_0 ... [SUCCESS]
> 2020-05-18 14:44:29,461 # 
> 2020-05-18 14:44:29,465 # Raw data T:  3056   V:   -43
> 2020-05-18 14:44:29,469 # Data Tabm: 2387   Tobj: 2711
> 2020-05-18 14:44:33,469 # Raw data T:  3052   V:   106
> 2020-05-18 14:44:33,472 # Data Tabm: 2384   Tobj: 3039
> 2020-05-18 14:44:37,471 # Raw data T:  3048   V:   110
> 2020-05-18 14:44:37,474 # Data Tabm: 2381   Tobj: 3045
> 2020-05-18 14:44:41,474 # Raw data T:  3044   V:   103
> 2020-05-18 14:44:41,477 # Data Tabm: 2378   Tobj: 3026
> 2020-05-18 14:44:45,476 # Raw data T:  3044   V:   120
> 2020-05-18 14:44:45,479 # Data Tabm: 2378   Tobj: 3063
> 2020-05-18 14:44:49,479 # Raw data T:  3036   V:   -70
> 2020-05-18 14:44:49,483 # Data Tabm: 2371   Tobj: 2633
> ```
> 
> I changed the conversion time to 4 seconds via `menuconfig`

Nice. Thanks for posting the test results.

**Comment by leandrolanzieri** [comment]
Looks good. Please squash.

**Comment by akshaim** [comment]
> Looks good. Please squash.

Done.

**Pull request merged**